src/store/zoom.ts

```diff
@@ -0,0 +1,151 @@
+	lastZoomTime: -1,
+};
+
+type UserEventAction = PayloadAction<{ zoomId: string; name: string }>;
+type UserLinkAction = PayloadAction<{ zoomId: string; discordId: string }>;
+
+const { reducer, actions, name } = createSlice({
+	name: 'zoomInfo',
+	reducers: {
+		callStarted: (state) => {
+			state.active = true;
+			state.online = [];
+			state.temporary = [];
+		},
+		callEnded: (state) => {
+			state.active = false;
+			state.hasSeenStart = true;
+			state.online = [];
+			// remove temp users
+			state.temporary.forEach((id) => {
+				delete state.byId[id];
+			});
+			state.temporary = [];
+		},
+		userLeft: (state, { payload }: UserEventAction) => {
+			// always go for the most up to date name
+			if (state.byId.hasOwnProperty(payload.zoomId)) {
+				state.byId[payload.zoomId].name = payload.name;
+			}
+
+			if (!state.active) {
+				console.log('WARNING: Zoom participant left while meeting was inactive. Ignoring.');
+			} else {
+				state.online = state.online.filter((zoomId) => zoomId !== payload.zoomId);
+				if (state.temporary.indexOf(payload.zoomId) > -1) {
+					state.temporary = state.temporary.filter((zoomId) => zoomId !== payload.zoomId);
+					delete state.byId[payload.zoomId];
+				}
+			}
+		},
+		userJoined: (
+			state,
+			{
+				payload: { temporary, ...payload },
+			}: PayloadAction<{ temporary: boolean }> & UserEventAction,
+		) => {
+			state.byId[payload.zoomId] = {
+				...state.byId[payload.zoomId],
+				...payload,
+			};
+			if (temporary) {
+				state.temporary.push(payload.zoomId);
+			}
+			if (!state.active) {
+				console.log(
+					'WARNING: Zoom participant joined while meeting was inactive. Ignoring.',
+				);
+			} else {
+				state.online.push(payload.zoomId);
+			}
+		},
+		userLinked: (state, { payload }: UserLinkAction) => {
+			state.byId[payload.zoomId].discordId = payload.discordId;
+		},
+		setJoinUrl: (state, { payload }: PayloadAction<string>) => {
+			state.joinUrl = payload;
+		},
+		setLastZoomTime: (state, { payload }: PayloadAction<number>) => {
+			state.lastZoomTime = payload;
+		},
+	},
+	initialState,
+});
+
+const getZoomInfo = (state: RootState): ZoomState => {
+	return state[name];
+};
+
+const getOnlineUsers = createSelector(getZoomInfo, (zoomInfo) =>
+	zoomInfo.online.map((id) => zoomInfo.byId[id]),
+);
+
+const getIsActive = createSelector(getZoomInfo, (zoomInfo) => zoomInfo.active);
+
+const getHasSeenStart = createSelector(getZoomInfo, (zoomInfo) => zoomInfo.hasSeenStart);
+
+const getJoinUrl = createSelector(getZoomInfo, (zoomInfo) => zoomInfo.joinUrl);
+
+const getLastZoomTime = createSelector(getZoomInfo, (zoomInfo) => zoomInfo.lastZoomTime);
+
+const getUserByName = (zoomName: string) =>
+	createSelector(getZoomInfo, (zoomInfo) => {
+		const match = Object.values(zoomInfo.byId).find(
+			({ name }) => name.toLowerCase() === zoomName.toLowerCase(),
+		);
+		if (match) {
+			return {
+				...match,
+				temporary: zoomInfo.temporary.indexOf(match.zoomId) > -1,
```

Comment:
I think it's better semantically to store temporary as a property of the user instead of as a separate array.

Reply:
I'm split on this. In this case (the selector) it does make it a bit messier. 

I'm slightly inclined to keep it separate because it is a category of user records and makes the cleanup of temporary users a lot cleaner. At the same time, because temp users are cleaned up at meeting end and we aren't dealing with a huge number of records, it doesn't feel as important to follow that convention.

Reply:
I see your point here, and I agree it could go either way.

In the abstract, I think it's better to design the model in such a way that it's impossible to represent invalid state. In this case, that would be that the `zoomInfo.temporary` array could contain a userID that doesn't exist, where as that's impossible if `temporary` is a property of the user.

That being said, that isn't a big difference here, and it's already written this way, so I don't think it's worth changing. Resolving this.